src/client.ts

```diff
@@ -80,151 +107,108 @@ export class DelphinusContract {
     }
     return { events: pastEvents, breakpoint: end };
   }
-
-  address() {
-    return this.contract.options.address;
-  }
 }
 
-export abstract class DelphinusWeb3 {
-  web3Instance: Web3;
-
-  constructor(web3Instance: Web3, close?: (_: provider) => Promise<void>) {
-    this.web3Instance = web3Instance;
+// Read only provider mode for node client (non-browser environment) when no private key is provided
+export class DelphinusReadOnlyProvider extends DelphinusProvider<DelphinusBaseProvider> {
```

Comment:
should we move all these Providers class to provider.ts?
I think for original web3 provider classes, they are all placed in provider.ts? Only WithXXXX is in client.ts?